homeassistant/components/kodi/media_player.py

```diff
@@ -327,8 +328,14 @@ def async_on_volume_changed(self, sender, data):
     @callback
     def async_on_quit(self, sender, data):
         """Reset the player state on quit action."""
+        self.hass.async_create_task(self._clear_connection())
+
+    @callback
+    async def _clear_connection(self, close=True):
```

Comment:
You can't have both `@callback` and `async def`. 

Also, why create task on not just execute this code directly inside `async_on_quit`?

Reply:
You're right, I'll remove the `@callback`. As for not running directly, `async_on_quit` is a callback from `jsonrpc-websocket` and I'm not 100% sure it's running in main HA loop.

Reply:
You need to find this out as you're calling an async function from the callback. That's not allowed if it's not from the event loop.

Reply:
Ok, so it is the main event loop (the loop is [taken from the session](https://github.com/emlove/jsonrpc-websocket/blob/80807df7ba2fdd9c3220254d36ce9a1a9735088c/jsonrpc_websocket/jsonrpc.py#L60) and we're passing the HA session).
Reading this again though, I'm not sure what's the advantage of running the code directly instead of creating a task? As you mentioned, eventually the code path leads to an async function, so at some point we need to create a task, and `_ _clear_connection()` is called from several places, so it makes sense to call it instead of repeating its code.